### ISSUE TITLE
Will nested Includes or IncludeAll() be re-included in v5?

### ISSUE THREAD

**Issue by srcmkr** [litedb-org/LiteDB#1280]
Hey Mauricio,

I was wondering, if there will be a re-include of the IncludeAll() like in v4? (or even better a nested Include w/ or w/o circular references)

My generic service function:
```
public T SingleOrDefault<T>(Expression<Func<T, bool>> expression)
{
	if (TenantExist(ConnectionString))
	{
		using (var db = new LiteDatabase(ConnectionString))
		{
			var collection = db.GetCollection<T>(typeof(T).Name.ToLower());
			return collection.IncludeAll().Find(expression).FirstOrDefault();
		}
	}
	else
		return default;
}
```

Thanks for your awesome work!

**Comment by srcmkr** [comment]
Update: Will the IncludeAll ever come back? atm it's a blocking feature for us to migrate.

**Comment by FooBartn** [comment]
@srcmkr have you tried using BsonExpression ?

**Comment by FooBartn** [comment]
Actually it looks like Parent.ChildArray[*].ChildArray[*] works

**Comment by srcmkr** [comment]
Indeed, the further levels are interesting, but one of the most challenging thing is the abstraction / generalization of the request.
We have too many models to create a separate service for each of them, so we started to use generic T.
Due to this, we can't pass or know the max depth, even ChildArray[*] works great.
Maybe we'll find a way to execute the ChildArray[*] recursivly, but the IncludeAll worked great so far.

**Comment by FooBartn** [comment]
Ah, gotcha. Totally understand -- you don't want to have to worry about property types / names. Hm. Well. I bet something could be done with reflection to build the BsonExpressions from property names and whether or not they're array types or not. 

Side note: I've noticed you seem to need to include every level:

```
.Include(BsonExpression.Create("$.Projects[*]"))
.Include(BsonExpression.Create("$.Projects[*].Scans[*]"))
.Include(BsonExpression.Create("$.Projects[*].Scans[*].NetDevices[*]"))
```

**Comment by EricVoll** [comment]
I would also love to see `IncludeAll` coming to v5. For us this is a blocking issue as well for the same reasons as for @srcmkr 
Are there any updates on this?

**Comment by lawrence-laz** [comment]
Still relevant. Is `IncludeAll` coming back?

**Comment by JKamsker** [comment]
Consolidated into #2778, which combines this report with the related ones and tracks the fix. Closing this as a duplicate — please follow #2778, and comment there if your case is different.
